src/util/MemorySize/MemorySize.h

```diff
@@ -145,38 +150,42 @@ constexpr static double sizeTDivision(const size_t dividend,
 }
 
 /*
-@brief Calculate the amount of bytes for a given amount of untis and a given
-amount of bytes per unit.
+@brief Calculate the amount of bytes for a given amount of units.
+
+@param unitName Must be `B`, `kB`, `MB`, `GB`, or `TB`.
 
 @return The amount of bytes. Rounded up, if needed.
 */
 template <typename T>
 requires std::integral<T> || std::floating_point<T>
 constexpr size_t convertMemoryUnitsToBytes(const T amountOfUnits,
-                                           const size_t numBytesPerUnit) {
+                                           std::string_view unitName) {
   // Negativ values makes no sense.
   AD_CONTRACT_CHECK(amountOfUnits >= 0);
 
+  // Must be one of the supported units.
+  AD_CONTRACT_CHECK(numBytesPerUnit.contains(unitName));
+
   // Max value for `amountOfUnits`.
-  if (static_cast<T>(sizeTDivision(size_t_max, numBytesPerUnit)) <
+  if (static_cast<T>(sizeTDivision(size_t_max, numBytesPerUnit.at(unitName))) <
       amountOfUnits) {
-    throw std::runtime_error(
-        "The given amount of memory is more memory, than can be represented "
-        "and worked with in c++.");
+    throw std::runtime_error(absl::StrCat(
+        amountOfUnits, " ", unitName,
+        " is more memory, than can be represented and worked with in c++."));
```

Comment:
This error message is not really English, let's think of something cleaner.
Maybe
" is larger than the maximum amount of memory that can be addressed using 64 bits".

Reply:
Done, see [4982fbb](https://github.com/ad-freiburg/qlever/pull/1032/commits/4982fbb1e0c5736444fed2a6b9b92925470c48ef).